1/ToDoApp/src/backend/resolvers/todo/todos.spec.js

```diff
@@ -0,0 +1,168 @@
+
+        });
+        describe('Modifying Todos', () => {
+            describe('given the loggedIn-User is the creator of the todo (allowed to modify)', () => {
+
+                it('updates todo message ', async () => {
+                    const res = await mutate({
+                        mutation: UPDATE_TODO_MESSAGE,
+                        variables: {
+                            id: 1,
+                            message: "kleines Update",
+                            finished: false,
+                            token: logInToken
+                        }
+                    });
+                    expect(res).toMatchObject({
+                        "data": {
+                            "updateTodo": {
+                                "message": "kleines Update"
+                            }
+                        }
+                    });
+                });
+                it('deletes todo ', async () => {
+                    const res = await mutate({
+                        mutation: DELETE_TODO,
+                        variables: {
+                            id: 1,
+                            token: logInToken
+                        }
+                    });
+                    expect(res.data.deleteTodo).toHaveLength(1);
+                });
+            });
+            describe('given the loggedIn-User is NOT the creator of the todo (NOT allowed to modify)', () => {
+                it('does not delete the todo', async () => {
+                    const res = await mutate({
+                        mutation: DELETE_TODO,
+                        variables: {
+                            id: 2,
+                            token: logInToken
+                        }
+                    });
+                    expect(res.data.deleteTodo.toBeUndefined);
+                });
+            });
+        });
+    });
+});
+
+const GET_ALL_TODOS = gql `
+                         query {
+                              allTodos {
+                                  id
+                                  message
+                                }
+                              }
+                            `;
+
+const GET_FIRST_TODOMESSAGE = gql `
+                    query {
+                      todoById(id:1) {
+                        message
+                      }
+                    }
+                  `;
+
+const CREATE_TODO = gql `
+                mutation addTodo($message: String, $token: String){
+                   addTodo(message: $message , token: $token)
+                    {
+                      message
+                    }
+                  }
+                `;
+
+const DELETE_TODO = gql `
+        mutation deleteTodo($id: Int, $token:String) {
+            deleteTodo(id: $id, token:$token)
+            {
+              id
+              message
+            }
+        }
+    `;
+
+const UPDATE_TODO_MESSAGE = gql `
```

Comment:
I would put each of these into the respective describe block for the mutation/query. If you want to be super concise, you can use a local variable called `mutation` or `query` which you would assign in a `beforeEach` and use it as input param for `query` and `mutate` respectively.

```js
let mutation
describe('updateTodo', () => {
  beforeEach(() => {
    mutation = gql `mutation updateTodo(...) { ... }`
  })
  it('some test case', () => {
    const expected = // something
    await expect(mutate({ mutation, variables })).resolves.toMatchObject(expected)
  })
})
```